distributed/shuffle/_shuffle.py

```diff
@@ -155,6 +155,20 @@ def __init__(
             annotations=annotations,
         )
 
+    def __reduce__(self) -> tuple:
+        attrs = [
+            "name",
+            "column",
+            "npartitions",
+            "npartitions_input",
+            "ignore_index",
+            "name_input",
+            "meta_input",
+            "parts_out",
+            "annotations",
+        ]
+        return (P2PShuffleLayer, tuple(getattr(self, attr) for attr in attrs))
```

Comment:
Can I ask why this was needed?

Reply:
I subclassed the `SimpleShuffleLayer` mostly to avoid code duplication back then. I believe with this change we could avoid the subclassing entirely.

The reason why this is necessary is because `SimpleShuffleLayer` implements the same reduction but with the `SimpleShuffleLayer` type. We could generalize this to use `type(self)` but I believe avoiding subclassing is the overall better approach